src/NzbDrone.Core/Download/RejectedImportService.cs

```diff
@@ -22,29 +22,35 @@ public RejectedImportService(ICachedIndexerSettingsProvider cachedIndexerSetting
 
     public bool Process(TrackedDownload trackedDownload, ImportResult importResult)
     {
-        if (importResult.Result != ImportResultType.Rejected || importResult.ImportDecision.LocalEpisode == null)
+        if (importResult.Result != ImportResultType.Rejected || importResult.ImportDecision.LocalEpisode == null || trackedDownload.RemoteEpisode?.Release == null)
         {
             return false;
         }
 
-        var indexerSettings = _cachedIndexerSettingsProvider.GetSettings(trackedDownload.RemoteEpisode.Release.IndexerId);
         var rejectionReason = importResult.ImportDecision.Rejections.FirstOrDefault()?.Reason;
 
+        if (trackedDownload.RemoteEpisode.Release.IndexerId == 0)
+        {
+            trackedDownload.Warn(new TrackedDownloadStatusMessage(importResult.Errors.First(), new List<string>()));
+            return true;
+        }
+
+        var indexerSettings = _cachedIndexerSettingsProvider.GetSettings(trackedDownload.RemoteEpisode.Release.IndexerId);
+
         if (rejectionReason == ImportRejectionReason.DangerousFile &&
             indexerSettings.FailDownloads.Contains(FailDownloads.PotentiallyDangerous))
         {
             trackedDownload.Fail();
         }
         else if (rejectionReason == ImportRejectionReason.ExecutableFile &&
-            indexerSettings.FailDownloads.Contains(FailDownloads.Executables))
+                 indexerSettings.FailDownloads.Contains(FailDownloads.Executables))
         {
             trackedDownload.Fail();
         }
         else
         {
             trackedDownload.Warn(new TrackedDownloadStatusMessage(importResult.Errors.First(), new List<string>()));
         }
-
```

Comment:
New line between the end of the block and return makes things less condensed and easier to read.